File: app-java/src/main/java/com/adforus/cubidsdksample/BaseSystemSetting.java
package com.adforus.cubidsdksample;

import android.os.Build;
import android.view.View;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class BaseSystemSetting {
    public static void setInsetDirection(View view) {
        if (Build.VERSION.SDK_INT >= 35) {
            ViewCompat.setOnApplyWindowInsetsListener(view, (v, insets) -> {
                Insets systemInsets = insets.getInsets(WindowInsetsCompat.Type.systemBars());
                Insets cutoutInsets = insets.getInsets(WindowInsetsCompat.Type.displayCutout());

                int topInset = Math.max(systemInsets.top, cutoutInsets.top);
                int bottomInset = Math.max(systemInsets.bottom, cutoutInsets.bottom);

                v.setPadding(
                        v.getPaddingLeft(),
                        v.getPaddingTop() + topInset,
                        v.getPaddingRight(),
                        v.getPaddingBottom() + bottomInset
                );

                return WindowInsetsCompat.CONSUMED;
            });
        }
    }
}

File: app/src/main/java/com/adforus/cubidsdksample/BaseSystemSetting.kt
package com.adforus.cubidsdksample

import android.view.View
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import kotlin.math.max

object BaseSystemSetting {
    fun setInsetDirection(view: View){
        if (android.os.Build.VERSION.SDK_INT >= 35) {
            ViewCompat.setOnApplyWindowInsetsListener(view) { view, insets ->
                val systemInsets = insets.getInsets(WindowInsetsCompat.Type.systemBars())
                val cutoutInsets = insets.getInsets(WindowInsetsCompat.Type.displayCutout())
                val topInset = max(systemInsets.top, cutoutInsets.top)
                val bottomInset = max(systemInsets.bottom, cutoutInsets.bottom)

                view.setPadding(view.paddingLeft, view.paddingTop + topInset, view.paddingRight, view.paddingBottom + bottomInset)
                WindowInsetsCompat.CONSUMED
            }
        }
    }
}